<?xml version="1.0"?>
<launch>
    <arg name="velo_expansion" default="0.3"/>
    <arg name="hokuyo_expansion" default="0.3"/>
    <node pkg="simple_localmap_creator" type="localmap_creator_node" name="localmap_creator_node" output="screen">
        <rosparam>
            hz: 10
            laser_names:
                - velodyne
                <!-- - front_hokuyo -->
            map_width: 20
            map_height: 20
            map_resolution: 0.1
        </rosparam>
        <remap from="/velodyne/scan" to="/scan"/>
        <!-- <remap from="/velodyne/scan" to="/recognition/road_recognizer/scan"/> -->
        <remap from="/localmap" to="/velodyne_map"/>
    </node>
    <!-- front lidar -->
    <node pkg="simple_localmap_creator" type="front_lidar_map" name="front_lidar_map" output="screen" >
        <param name="map_frame" value="base_link"/>
        <param name="map_width" value="200.0"/>
        <param name="map_height" value="200.0"/>
        <param name="map_resolution" value="0.1"/>
        <param name="expansion_rate" value="$(arg hokuyo_expansion)"/>
        <!-- <remap from="front_lidar_map" to="hokuyo_map"/> -->
    </node>
    <!-- merge  -->
    <node pkg="simple_localmap_creator" type="merged_map_creator" name="merged_map_creator" >
        <param name="map_frame" value="base_link"/>
        <param name="map_size" value="20.0"/>
        <param name="map_resolution" value="0.1"/>
        <param name="expansion_rate" value="$(arg velo_expansion)"/>
        <param name="map_min_range" value="0.0"/>
        <param name="map_topic1" value="/velodyne_map"/>
        <param name="map_topic2" value="/front_lidar_map"/>
        <remap from="/merged_local_map" to="/local_map/expand"/>
    </node>
    <node pkg="simple_localmap_creator" type="road_projector" name="road_projector" output="screen">
        <remap from="/localmap" to="/local_map/expand"/>
        <remap from="/road" to="/road"/>
        <remap from="/projected_map" to="/localmap/projected_roadedge"/>
        <param name="visualize" value="true"/>
        <param name="road_thickness" value="0.3"/>
        <param name="ignore_wall_radius" value="2.0"/>
        <param name="robot_length" value="0.5"/>
    </node>
    <!-- tf -->
<!-- <node pkg="tf2_ros" type="static_transform_publisher" name="static_transform_publisher_velodyne" output="screen" args="0 0 0 0 0 0 base_link velodyne"/> -->
<!-- <node pkg="tf2_ros" type="static_transform_publisher" name="static_transform_publisher_hokuyo" output="screen" args="0 0 0 0 0 0 base_link front_hokuyo"/> -->

</launch>
